posthog/models/product_intent/product_intent.py

```diff
@@ -74,12 +80,47 @@ def has_activated_data_warehouse(self) -> bool:
 
         return False
 
-    def check_and_update_activation(self) -> None:
-        if self.product_type == "data_warehouse":
-            if self.has_activated_data_warehouse():
-                self.activated_at = datetime.now(tz=UTC)
-                self.save()
-                self.report_activation("data_warehouse")
+    def has_activated_experiments(self) -> bool:
+        # the team has any launched experiments
+        return Experiment.objects.filter(team=self.team, start_date__isnull=False).exists()
+
+    def has_activated_feature_flags(self) -> bool:
+        # Get feature flags that have at least one filter group
+        feature_flags = (
+            FeatureFlag.objects.filter(
+                team=self.team,
+                filters__groups__0__properties__isnull=False,
+            )
+            .exclude(name__istartswith="Feature Flag for Experiment")
+            .exclude(name__istartswith="Targeting flag for survey")
```

Comment:
Hm, that's probably a better solution. It shouldn't be too difficult because the discrete number of flags / experiments per project is low (ie in the hundreds, not hundreds of thousands).

I'll do that.